File: app/src/test/java/com/example/benchmarks/MainThreadRule.java
package com.example.benchmarks;

import androidx.annotation.NonNull;

import org.junit.rules.TestWatcher;
import org.junit.runner.Description;

import io.reactivex.rxjava3.android.plugins.RxAndroidPlugins;
import io.reactivex.rxjava3.plugins.RxJavaPlugins;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class MainThreadRule extends TestWatcher {

    @Override
    protected void starting(@NonNull Description description) {
        RxAndroidPlugins.setInitMainThreadSchedulerHandler(scheduler -> Schedulers.trampoline());
        RxJavaPlugins.setComputationSchedulerHandler(scheduler -> Schedulers.trampoline());
        super.starting(description);
    }

    @Override
    protected void finished(@NonNull Description description) {
        RxAndroidPlugins.reset();
        RxJavaPlugins.reset();
        super.finished(description);
    }
}

File: app/src/test/java/com/example/benchmarks/MainThreadRule.kt
package com.example.benchmarks

import io.reactivex.rxjava3.android.plugins.RxAndroidPlugins
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.plugins.RxJavaPlugins
import io.reactivex.rxjava3.schedulers.Schedulers
import org.junit.rules.TestWatcher
import org.junit.runner.Description
import java.util.concurrent.Callable


open class MainThreadRule : TestWatcher() {

    override fun starting(description: Description) {
        RxAndroidPlugins.setInitMainThreadSchedulerHandler { scheduler: Callable<Scheduler?>? -> Schedulers.trampoline() }
        RxJavaPlugins.setComputationSchedulerHandler { Schedulers.trampoline() }
        super.starting(description)
    }

    override fun finished(description: Description) {
        RxAndroidPlugins.reset()
        RxJavaPlugins.reset()
        super.finished(description)
    }

}